provider/metrics/metrics.go

```diff
@@ -0,0 +1,223 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+	"github.com/pkg/errors"
+	"github.com/virtual-kubelet/azure-aci/client/aci"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	stats "github.com/virtual-kubelet/virtual-kubelet/node/api/statsv1alpha1"
+	"github.com/virtual-kubelet/virtual-kubelet/trace"
+	"golang.org/x/sync/errgroup"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	ContainerGroupCacheTTLSeconds = 60 * 5
+)
+
+// package dependency: query the Pods in current virtual nodes. it usually is ResourceManager
+type PodGetter interface {
+	GetPods() []*v1.Pod
+}
+
+// package dependency: query the Pod's correspoinding Container Group metrics from Container Insights
+type ContainerGroupMetricsGetter interface {
```

Comment:
`MetricsGetter`

Reply:
As I mentioned before. The longer name should be more meaningful and easier for reading by some else.

Reply:
Am I not someone else?
For me, the code is wordy enough that I'm having a difficult time to get through it (I have reading issues in general).
Also, these objects/interfaces are already within the context of this metrics package within the ACI provider.

That said I'm only making suggestions but am definitely not nit-picking.

Reply:
agree with you. I fixed that.